server/db/schema/hostels.schema.ts

```diff
@@ -0,0 +1,48 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, text, timestamp, varchar } from 'drizzle-orm/pg-core';
+
+import { faculty, staff } from '.';
+
+export const hostels = pgTable('hostels', {
+  id: serial('id').primaryKey(),
+  address: text('address').notNull(),
+  name: varchar('name', { length: 256 }).notNull(),
+  urlName: varchar('url_name', { length: 128 }).notNull(),
+  email: varchar('email', { length: 256 }).notNull(),
+  telephone: varchar('telephone', { length: 13 }).notNull(),
+  alternateTelephone: varchar('alternate_telephone', { length: 13 }),
+  type: varchar('type', {
+    enum: ['boys', 'girls'],
+  }).notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  updatedAt: timestamp('updated_at')
+    .$onUpdate(() => new Date())
+    .notNull(),
+});
+
+export const hostelsRelations = relations(hostels, ({ many }) => ({
+  staff: many(hostelStaff),
+  wardens: many(faculty, { relationName: 'wardens' }),
+  deputyWardens: many(faculty, { relationName: 'deputy-wardens' }),
```

Comment:
If you have a `many` relationship, you need to have a `one` relationship somewhere else. See `doctorates` in `faculty` and `students` for example.

Reply:
@KambojRajan suggested that it's not required.

Reply:
Did you see the usage? Are you able to fetch a hsotel with its wardens appropriately?

Reply:
Sorry Im facing some problems with it. this was working as it is before now. It doesnt even work with one relation in faculty, but somehow if i reference different fields it starts working.